Makefile.am

```diff
@@ -2,6 +2,10 @@ AUTOMAKE_OPTIONS=foreign
 
 AM_CFLAGS=-pedantic -Wall -Werror -Wextra
 
+if LINUX
```

Comment:
I don't see why, from a quick skim of the repo, `pledge` disappears when defining `_XOPEN_SOURCE`. But I do see this in glibc: https://github.com/bminor/glibc/blob/963394a22b38c4ec92b6875a6c06d3b15d5c0d21/include/features.h#L186-L208

So maybe: skip the OS check (which should be a glibc check ... somehow?) and instead do:

```
AM_CPPFLAGS+=-D_GNU_SOURCE
```